aubio-rs/src/filterbank.rs

```diff
@@ -0,0 +1,141 @@
+use crate::{
+    check_init,
+    Error::{InvalidArg},
+    ffi,
+    vec::{FVecMut, CVec, FMat, FMatVecs},
+    Result,
+    Status,
+};
+
+/**
+ * Filterbank object.
+ *
+ * General-purpose spectral filterbank object.
+ */
+pub struct FilterBank {
+    filterbank: *mut ffi::aubio_filterbank_t,
+    n_filters: usize,
+    win_s: usize,
+}
+
+impl Drop for FilterBank {
+    fn drop(&mut self) {
+        unsafe { ffi::del_aubio_filterbank(self.filterbank) }
+    }
+}
+
+impl FilterBank {
+    /**
+     * Create filterbank object.
+     *
+     * - `n_filters`: Number of filters to create
+     * - `win_s`: size of analysis buffer (and length of the FFT transform)
+     *
+     * Allocates an empty matrix of length win_s / 2 + 1 and height n_filters
+     */
+    pub fn new(n_filters: usize, win_s: usize) -> Result<Self> {
+        let filterbank = unsafe {
+            ffi::new_aubio_filterbank(n_filters as ffi::uint_t, win_s as ffi::uint_t)
+        };
+
+        check_init(filterbank)?;
+
+        Ok(Self { filterbank, n_filters, win_s })
+    }
+
+    pub fn set_coeffs(&mut self, filters: FMat<FMatVecs>) -> Status {
+        if filters.height() != self.n_filters || filters.length() != self.win_s / 2 + 1 {
```

Comment:
I think that bindings does not add extra overhead by default. But since aubio itself neglects checks, for things like this I introduced optional feature "check-size".

See [Cargo.toml](https://github.com/katyo/aubio-rs/blob/94dabac8c453d23fb7011eda3f71ee1c8af4c3be/aubio-rs/Cargo.toml#L34).

You need to add `#[cfg(feature = "check-size")]{ /* your code which check sizes */ }`.